src/MultiVariablePolynomial.java

```diff
@@ -0,0 +1,57 @@
+import java.util.LinkedList;
+public class MultiVariablePolynomial
+{
+    String str;
+    LinkedList<LinkedList<Integer>> list = new LinkedList<LinkedList<Integer>>();
+
+    MultiVariablePolynomial(String str)
+    {
+        this.str = str;
+    }
+
+    int calculateMaxDegree()
+    {
+        int max = Integer.MIN_VALUE;
+        processLinkedList();
```

Comment:
This should not be part of calculatingMaxDegree. So either you could move it into cunstructor or into init mehtod.

Reply:
done